core/config/v2/docs/docs_test.go

```diff
@@ -55,6 +55,9 @@ func TestDoc(t *testing.T) {
 		require.Equal(t, ks, docDefaults.KeySpecific[0])
 		docDefaults.KeySpecific = nil
 
+		// EVM.GasEstimator.BumpTxDepth defaults to EVM.Transactions.MaxInFlight
+		docDefaults.GasEstimator.BumpTxDepth = nil
```

Comment:
Should we be checking for a zero value like the others?
How about documenting like `EIP1559FeeCapBufferBlocks`, below on line 66/69?
> // EIP1559FeeCapBufferBlocks doesn't have a constant default - it is derived from another field

Reply:
If the value is set to 0, all tx will be bumped - i.e. the 0 value is meaningful. So in this case, I don't think we should have similar behavior to the other fields (like on line 59/62). 

I think the comment on line 58 of this revision is clearer than what is on line 66/69, but I can copy the syntax.

Reply:
Sorry, I'm not following. The purpose of the `require.*` checks inside of this subtest are to ensure the values loaded from the default before niling them out in order to run `assertTOML(` cleanly. We need to be checking `docDefaults.GasEstimator.BumpTxDepth` for _something_ here, otherwise we risk a false positive if this changes in the future. IIUC it will have a zero value, which happens to be literal `0` in this case, but it shouldn't matter if that is meaningful - that's just how fields w/o defaults get filled in.

Reply:
Yep, that's right. I edited my original comment after taking a deeper look.

Reply:
Ah, ok - thanks for providing the context on the intent around setting the 0 values.

I think that's a fine way of doing this, but it would be clearer to me if there was a separate unit test for regression testing on what config fields have a defaults vs don't.

Reply:
Yeah this has grown a lot. The original point was that none should be nil, but we've accumulated many exceptions.